src/compas/files/base_reader.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+
+    def __init__(self, location):
+        self.is_url = location
+        self.location = location
+
+    @property
+    def is_url(self):
+        return self._is_url
+
+    @is_url.setter
+    def is_url(self, location):
+        """Determine if location is local or specified as an URL
+        """
+        if str(location).startswith('http'):
+            self._is_url = True
+        else:
+            self._is_url = False
+
+    @property
+    def location(self):
+        return self._location
+
+    @location.setter
+    def location(self, location):
+        """Ensure that location is specified using Path object
+        """
+        if self.is_url:
+            self._location = location
+        else:
+            if not isinstance(location, Path):
+                pathobj = Path(location)
+            else:
+                pathobj = location
+
+            if pathobj.exists():
+                self._location = pathobj
+            else:
+                raise IOError('File not found.')
+
+    def read_line_or_chunk(self, mode='ascii'):
+        """Returns iterable reading file line by line
+
+        Should accept both local files as well as URLs, in binary or ascii format.
+
+        Parameters
+        ---------
+        mode : string
+            Treat file as ascii or binary
+        Yields
+        -------
+        string
+            Next line or chunk of file
+        """
+        # TODO: Handle url's to binary files
+        # TODO: Handle continuing lines (as in OFF files)
+        # TODO: Read binary chunks
+
+        self.check_file_signature()
+
+        if mode == 'ascii':
+            print(self.is_url)
+
+            if self.is_url:
+                resp = urlopen(self.location)
+                for line in resp.read().decode('utf-8').split('\n'):
+                    yield line
+            else:
+                try:
+                    with self.location.open(mode='r') as fh:
+                        for line in fh:
+                            yield line
+                except UnicodeDecodeError:
+                    with self.location.open(mode='r', errors='replace', newline='\r') as fh:
```

Comment:
should this not be moved to `open`?

Reply:
`_open` so there's no conflict. Moved it out of `iter_lines`.

Reply:
Created open_ascii and open_binary, only difference is `mode='r'` vs `mode='rb`.